examples/optionals/as_bool.rs

```diff
@@ -0,0 +1,152 @@
+#![deny(warnings)]
+#![warn(rust_2018_idioms)]
+
+#[macro_use]
+extern crate log;
+
+use serde::Deserialize;
+use serde_this_or_that::as_opt_bool;
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Msg {
+    #[serde(deserialize_with = "as_opt_bool")]
+    pub timestamp: Option<bool>,
+}
+
+// A simple type alias so as to DRY.
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+fn main() -> Result<()> {
+    sensible_env_logger::init!();
+
+    trace!("With Empty String:");
+    let data = r#"
+    {
+        "timestamp": ""
+    }"#;
+
+    let m: Msg = serde_json::from_str(data).unwrap();
+    assert_eq!(m.timestamp, Some(false));
```

Comment:
This should be `None`, shouldn't it? When parsing into an Option, only unambiguiously parseable strings should be turned into a Some variant. At least that would be my expectation. Or do we follow Javascript parsing logic, that is much laxer and casts eagerly?
```suggestion
    assert_eq!(m.timestamp, None);
```

Maybe we should write out all the edge cases and how they should treat it.

Reply:
Done @corneliusroemer ! Thanks for bringing this up. I added a check for all "falsy" string values explicitly. All other (invalid) string values should be deserialized to `None` as suggested.